sdks/java/core/src/main/java/org/apache/beam/sdk/util/RowJsonUtils.java

```diff
@@ -51,21 +51,20 @@ public class RowJsonUtils {
    * overwrite the default buffer size limit to 100 MB, and exposes this interface for higher limit.
    * If needed, call this method during pipeline run time, e.g. in DoFn.setup.
    */
-  public static void increaseDefaultStreamReadConstraints(int newLimit) {
-    if (newLimit <= defaultBufferLimit) {
-      return;
+  public static synchronized void increaseDefaultStreamReadConstraints(int newLimit) {
+    if (newLimit > defaultBufferLimit) {
+      try {
+        Class.forName("com.fasterxml.jackson.core.StreamReadConstraints");
```

Comment:
we can check STREAM_READ_CONSTRAINTS_AVAILABLE then return early and remove this try {} block